### PR TITLE
Basis for projectiles and revamped main

### PR DESCRIPTION
**Pull request by Daskie**
Projectiles are essentially just a collection of pre-existing components. As of now left click just "shoots" a  burger that has gravity and collides with the scene, dissapearing and fattening any enemy it hits. 
Additionally reworked main to be way more organized and provide a basis for expansion.

### CODE REVIEW

File: src/Engine/Component/EnemyComponents/EnemyComponent.cpp [hunk 1]
```diff
@@ -0,0 +1 @@
+#include "EnemyComponent.hpp"
```
Comment (jaafersheriff) [inline src/Engine/Component/EnemyComponents/EnemyComponent.cpp]: TODO?

File: src/Engine/Loader/Loader.cpp [hunk 2]
```diff
@@ -114,8 +114,8 @@ Texture* Loader::getTexture(const String & name) {
     return getTexture(name, GL_REPEAT, true);
 }
 
-int Loader::loadLevel(const String & name) {
-    return FileReader::loadLevel(*name.c_str());
+int Loader::loadLevel(const String & name, float ambience) {
```
Comment (jaafersheriff) [inline src/Engine/Loader/Loader.cpp]: Shouldn't ambience be per material?

File: src/Engine/Component/EnemyComponents/EnemyComponent.cpp (hunk 1 above)
Comment (Daskie) [reply src/Engine/Component/EnemyComponents/EnemyComponent.cpp]: yes

File: src/Engine/Loader/Loader.cpp (hunk 2 above)
Comment (Daskie) [reply src/Engine/Loader/Loader.cpp]: yes, but we're not importing any external material properties at the moment. Until Zack does that, if he does that, i figured it was important to have one global ambience value defined in main.

Comment (jaafersheriff) [reply src/Engine/Loader/Loader.cpp]: @zackragozzino

**Pull request merged**